src/Sulu/Component/Util/SortUtils.php

```diff
@@ -96,4 +96,49 @@ public static function multisort($values, $paths, $direction = 'ASC')
 
         return $values;
     }
+
+    /**
+     * Sorts the items of the iterable by a locale aware function.
+     *
+     * The values for comparison are casted to string, before they are compared. The sorted items are returned by a new array.
+     *
+     * If the intl extension is not loaded, the comparison falls back to binary comparison.
+     *
+     * @param iterable<mixed> $list
+     * @param null|callable $getComparableValue callback to get the value from each item that will be compared
+     *
+     * @return mixed[]
+     *
+     * @throws \InvalidArgumentException if the comparison of the values failed
+     */
+    public static function sortLocaleAware(iterable $list, string $locale, ?callable $getComparableValue = null): array
+    {
+        $array = \is_array($list) ? $list : \iterator_to_array($list);
+
+        if (null === $getComparableValue) {
+            $getComparableValue = fn ($item) => $item;
+        }
+
+        // Collator class requires intl extension
+        $collator = \class_exists(\Collator::class) ? new \Collator($locale) : null;
+
+        \usort($array, function(mixed $itemA, mixed $itemB) use ($collator, $getComparableValue) {
```

Comment:
I think the util function should keep the index:

```suggestion
        \uasort($array, function(mixed $itemA, mixed $itemB) use ($collator, $getComparableValue) {
```